opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/LuceneQuery.java

```diff
@@ -32,10 +32,13 @@
 import org.opensearch.sql.expression.ReferenceExpression;
 import org.opensearch.sql.expression.function.BuiltinFunctionName;
 import org.opensearch.sql.expression.function.FunctionName;
+import org.opensearch.sql.opensearch.data.type.OpenSearchDateType;
 
 /** Lucene query abstraction that builds Lucene query from function expression. */
 public abstract class LuceneQuery {
 
+  private ReferenceExpression ref;
```

Comment:
Maybe you could add this field and an override method `build()` in `SingleFieldQuery`, instead of `LuceneQuery`. My another concern is will this class be used in multiple threads? Adding a field in XXQuery will cause bug in multiple threads case.